# [문제1]

다음의 Series를 생성하고 sample 변수에 대입하고 출력하는 코드를 작성하시오.

[출력]

가    10

나    20

다    30

라    40

마    50

dtype: int64

In [24]:
import pandas as pd

sample = pd.Series([10, 20, 30, 40, 50], index=["가", "나", "다", "라", "마"])
print(sample)

가    10
나    20
다    30
라    40
마    50
dtype: int64


# [문제2]

아래의 코드를 사용하여 sample2 중 160 이하인 데이터만 필터링여 출력하는  코드를 작성하시오.

np.random.seed(20)
    
sample2 = pd.Series(np.random.randint(100, 200, size=(15,)))

sample2

[출력]

2     115

4     128

6     109

7     120

9     122

11    134

13    140

dtype: int64

In [39]:
import numpy as np

np.random.seed(20)
sample2 = pd.Series(np.random.randint(100, 200, size=(15,)))
print(sample2[sample2<=160])

2     115
4     128
6     109
7     120
9     122
11    134
13    140
dtype: int32


# [문제3]
다음과 같은 Series를 생성하는 코드를 작성하시오.

[출력]

가     apple

나       NaN

다    banana

라      kiwi

마    gubong

dtype: object

In [26]:
sample = sample.astype("object")
sample[:] = ["apple", np.nan, "banana", "kiwi", "gubong"]
print(sample)

가     apple
나       NaN
다    banana
라      kiwi
마    gubong
dtype: object


# [문제4]
위 예제에서 결측치가 아닌 데이터만 필터링하여 출력하는 코드를 작성하시오.

[출력]

가     apple

다    banana

라      kiwi

마    gubong

dtype: object

In [27]:
a = sample[sample.notnull()]
print(a)

가     apple
다    banana
라      kiwi
마    gubong
dtype: object


# [문제5]
위 예제에서 결측치 데이터만 필터링하여 출력하는 코드를 작성하시오.

[출력]

나       NaN

In [28]:
b = sample[sample.isnull()]
print(b)

나    NaN
dtype: object


# [문제6]
아래 코드를 사용하여 sample에서 다음과 같은 결과를 가지도록 슬라이싱하여 출력하는 코드를 작성하시오.

np.random.seed(0)

sample = pd.Series(np.random.randint(100, 200, size=(10,)))

sample

[출력]

2    164

3    167

4    167

5    109

6    183

dtype: int64

In [34]:
np.random.seed(0)
sample = pd.Series(np.random.randint(100, 200, size=(10,)), index=range(2, 12))
sample

2     144
3     147
4     164
5     167
6     167
7     109
8     183
9     121
10    136
11    187
dtype: int32

# [문제7]
아래 코드를 사용하여 sample2에서 다음과 같은 결과를 가지도록 슬라이싱하여 출력하는 코드를 작성하시오.((1)~(3)번 세문제임)

np.random.seed(0)

sample2 = pd.Series(np.random.randint(100, 200, size=(10,)), index=list('가나다라마바사아자차'))

sample2

(1) 
[출력]

바    109

사    183

아    121

자    136

차    187

dtype: int64

(2)

[출력]

가    144

나    147

다    164

dtype: int64

(3)

[출력]

나    147

다    164

라    167

마    167

바    109

dtype: int64

In [41]:
np.random.seed(0)

sample2 = pd.Series(np.random.randint(100, 200, size=(10,)), index=list('가나다라마바사아자차'))

print(sample2[5:])
print(sample2[:3])
print(sample2[1:6])

바    109
사    183
아    121
자    136
차    187
dtype: int32
가    144
나    147
다    164
dtype: int32
나    147
다    164
라    167
마    167
바    109
dtype: int32
